src/Storefront/Controller/CartLineItemController.php

```diff
@@ -3,6 +3,11 @@
 namespace Shopware\Storefront\Controller;
 
 use Shopware\Core\Checkout\Cart\Cart;
+use Shopware\Core\Checkout\Cart\Event\CartLineItemAddedEvent;
```

Comment:
Firing the events in the controller is superfluous, since this has been built with the [store API](https://docs.shopware.com/en/shopware-platform-dev-en/developer-guide/controller#store-api-controller-route). This means every service call here in turn calls the store API route and therefore already triggers the events.